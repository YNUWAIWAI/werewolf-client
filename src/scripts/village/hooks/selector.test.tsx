import * as React from 'react'
import {mount} from 'enzyme'
import {useSelector} from './selector'

interface Props {
  len: number
}
const Test: React.FC<Props> = props => {
  const selector = useSelector(0)

  React.useEffect(() => {
    selector.setLength(props.len)
  }, [props.len, selector])
  const prev = 'prev'
  const next = 'next'
  const refresh = 'refresh'

  return (
    <div>
      <span>
        {selector.selectedItem}
      </span>
      <button
        id={prev}
        onClick={() => selector.prev()}
      >
        {prev}
      </button>
      <button
        id={next}
        onClick={() => selector.next()}
      >
        {next}
      </button>
      <button
        id={refresh}
        onClick={() => selector.refresh()}
      >
        {refresh}
      </button>
    </div>
  )
}

Test.displayName = 'Test'

test('render', () => {
  const wrapper = mount(
    <Test
      len={5}
    />
  )

  expect(wrapper.find('span').text()).toBe('0')
  expect(wrapper.html()).toMatchSnapshot()
})
test('next', () => {
  const wrapper = mount(
    <Test
      len={5}
    />
  )

  wrapper.find('#next').simulate('click')
  expect(wrapper.find('span').text()).toBe('1')
  expect(wrapper.html()).toMatchSnapshot()
})
test('prev', () => {
  const wrapper = mount(
    <Test
      len={5}
    />
  )

  wrapper.find('#prev').simulate('click')
  expect(wrapper.find('span').text()).toBe('4')
  expect(wrapper.html()).toMatchSnapshot()
})
test('refresh', () => {
  const wrapper = mount(
    <Test
      len={5}
    />
  )

  wrapper.find('#next').simulate('click')
  wrapper.find('#next').simulate('click')
  expect(wrapper.find('span').text()).toBe('2')
  wrapper.find('#refresh').simulate('click')
  expect(wrapper.find('span').text()).toBe('0')
  expect(wrapper.html()).toMatchSnapshot()
})
