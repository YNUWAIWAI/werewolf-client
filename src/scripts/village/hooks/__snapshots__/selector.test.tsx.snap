// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`next 1`] = `"<div><span>1</span><button id=\\"prev\\">prev</button><button id=\\"next\\">next</button><button id=\\"refresh\\">refresh</button></div>"`;

exports[`prev 1`] = `"<div><span>4</span><button id=\\"prev\\">prev</button><button id=\\"next\\">next</button><button id=\\"refresh\\">refresh</button></div>"`;

exports[`refresh 1`] = `"<div><span>0</span><button id=\\"prev\\">prev</button><button id=\\"next\\">next</button><button id=\\"refresh\\">refresh</button></div>"`;

exports[`render 1`] = `"<div><span>0</span><button id=\\"prev\\">prev</button><button id=\\"next\\">next</button><button id=\\"refresh\\">refresh</button></div>"`;
